src/drivers/spi/bcm283x/bcm283x_spi0.c

```diff
@@ -0,0 +1,217 @@
+/**
+ * @file bcm283x_spi0.c
+ * @brief 
+ * @author kpishere
+ * @version 0.1
+ * @date 2021.07.16
+ */
+#include <errno.h>
+
+#include <drivers/common/memory.h>
+#include <drivers/spi.h>
+#include <embox/unit.h>
+#include <framework/mod/options.h>
+#include <hal/reg.h>
+#include <util/log.h>
+#include <util/math.h>
+#include <kernel/irq.h>
+#include <kernel/printk.h>
+#include <asm/delay.h>
+
+#include <drivers/gpio/gpio.h>
+#include <drivers/gpio/bcm283x/bcm283x_gpio.h>
+//#include <drivers/mailbox/bcm2835_mailbox_property.h>
+#include <drivers/spi.h>
+
+#include "bcm283x_spi0.h"
+
+#define PBASE		OPTION_GET(NUMBER,base_addr)
+#define AUXBASE		OPTION_GET(NUMBER,aux_addr)
+
+#define SPI_BUS_CLOCK_DIVISOR 	OPTION_GET(NUMBER, spi_bus_clock_divisor)
+
+// Toggle an extra GPIO pin at various points to debug/test timing
+#if 0
```

Comment:
Maybe it is better to use a option from Mybuild for this purpose?

Reply:
Considered that but really, the meaning of this pin being set is only understood by careful code inspection, it is there only for performance/behaviour testing.  It could be removed and/or change from version to version depending on what is being actively developed/tested.  No functional or integration with other modules kind if value either so, maybe not?  Unless there are other/more reasons to put there?

Reply:
Ok, I see and agree with you. It is more convenient for this case to use a macro definition.